static-assets/js/tinymce-plugins/template/plugin.js

```diff
@@ -23,387 +23,422 @@
-          n.innerHTML = getDateTime(editor, getMdateFormat(editor));
-        }
-        if (hasAnyClasses(dom, n, getSelectedContentClasses(editor))) {
-          n.innerHTML = sel;
+      html = html.replace(new RegExp('\\{\\$' + escape(k) + '\\}', 'g'), v);
+    });
+    return html;
+  };
+  const replaceVals = (editor, scope) => {
+    const dom = editor.dom,
+      vl = getTemplateReplaceValues(editor);
+    global$2.each(dom.select('*', scope), (e) => {
+      global$2.each(vl, (v, k) => {
+        if (dom.hasClass(e, k)) {
+          if (isFunction(v)) {
+            v(e);
+          }
         }
       });
-      replaceVals(editor, el);
-      editor.execCommand('mceInsertContent', false, el.innerHTML);
-      editor.addVisual();
-    };
+    });
+  };
+  const insertTemplate = (editor, _ui, html) => {
+    const dom = editor.dom;
+    const sel = editor.selection.getContent();
+    html = replaceTemplateValues(html, getTemplateReplaceValues(editor));
+    let el = dom.create('div', {}, parseAndSerialize(editor, html));
+    const n = dom.select('.mceTmpl', el);
+    if (n && n.length > 0) {
+      el = dom.create('div');
+      el.appendChild(n[0].cloneNode(true));
+    }
+    global$2.each(dom.select('*', el), (n) => {
+      if (hasAnyClasses(dom, n, getCreationDateClasses(editor))) {
+        n.innerHTML = getDateTime(editor, getCdateFormat(editor));
+      }
+      if (hasAnyClasses(dom, n, getModificationDateClasses(editor))) {
+        n.innerHTML = getDateTime(editor, getMdateFormat(editor));
+      }
+      if (hasAnyClasses(dom, n, getSelectedContentClasses(editor))) {
+        n.innerHTML = sel;
+      }
+    });
+    replaceVals(editor, el);
+    editor.execCommand('mceInsertContent', false, el.innerHTML);
+    editor.addVisual();
+  };
 
-    var global = tinymce.util.Tools.resolve('tinymce.Env');
+  var global = tinymce.util.Tools.resolve('tinymce.Env');
 
-    const getPreviewContent = (editor, html) => {
-      var _a;
-      let previewHtml = parseAndSerialize(editor, html);
-      if (html.indexOf('<html>') === -1) {
-        let contentCssEntries = '';
-        const contentStyle = (_a = getContentStyle(editor)) !== null && _a !== void 0 ? _a : '';
-        const cors = shouldUseContentCssCors(editor) ? ' crossorigin="anonymous"' : '';
-        global$2.each(editor.contentCSS, url => {
-          contentCssEntries += '<link type="text/css" rel="stylesheet" href="' + editor.documentBaseURI.toAbsolute(url) + '"' + cors + '>';
+  const getPreviewContent = (editor, html) => {
+    var _a;
+    let previewHtml = parseAndSerialize(editor, html);
+    if (html.indexOf('<html>') === -1) {
+      let contentCssEntries = '';
+      const contentStyle = (_a = getContentStyle(editor)) !== null && _a !== void 0 ? _a : '';
+      const cors = shouldUseContentCssCors(editor) ? ' crossorigin="anonymous"' : '';
+      const templatesCSS = editor.options.get('templates_css') ?? [];
```

Comment:
@rart prettier ended up formatting the whole file, though the changes are lines 383 and 384. Please let me know if you want a non-prettified version of the file only with the changes